File: frontend/src/types/feedback.ts
export interface FeedbackCreate {
  session_id: string;
  message_id: string;
  question: string;
  answer: string;
  route_selected?: string;
  feedback_type: 'helpful' | 'not_helpful';
  feedback_comment?: string;
}

export interface FeedbackResponse {
  id: number;
  user_id: number | null;
  session_id: string;
  message_id: string;
  question: string;
  answer: string;
  route_selected: string | null;
  feedback_type: string;
  feedback_comment: string | null;
  created_at: string;
}

export interface FeedbackStats {
  total: number;
  helpful: number;
  not_helpful: number;
  helpful_pct: number;
  not_helpful_pct: number;
  by_agent: Record<string, { helpful: number; not_helpful: number }>;
}

File: backend/app/schemas/feedback.py
"""
Pydantic schemas for feedback request/response validation.
"""

from datetime import datetime

from pydantic import BaseModel


class FeedbackCreate(BaseModel):
    """Schema for submitting new feedback."""
    session_id: str
    message_id: str
    question: str
    answer: str
    route_selected: str | None = None
    feedback_type: str  # "helpful" or "not_helpful"
    feedback_comment: str | None = None


class FeedbackResponse(BaseModel):
    """Schema for returning feedback records."""
    id: int
    user_id: int | None
    session_id: str
    message_id: str
    question: str
    answer: str
    route_selected: str | None
    feedback_type: str
    feedback_comment: str | None
    created_at: datetime

    class Config:
        from_attributes = True


class FeedbackStats(BaseModel):
    """Aggregated feedback statistics for the admin analytics page."""
    total: int
    helpful: int
    not_helpful: int
    helpful_pct: float
    not_helpful_pct: float
    by_agent: dict[str, dict[str, int]]
